x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -225,9 +226,8 @@ export class Authenticator {
     );
     this.serverBasePath = this.options.basePath.serverBasePath || '/';
 
-    // only set these vars if they are defined in options (otherwise coalesce to existing/default)
-    this.idleTimeout = this.options.config.session.idleTimeout;
-    this.lifespan = this.options.config.session.lifespan;
+    this.idleTimeout = durationToMs(this.options.config.session.idleTimeout);
```

Comment:
__question:__ any reason we don't want to keep `Duration` type and only convert to `ms` when we really need it? `Duration` provides [a number of convenient methods](https://momentjs.com/docs/#/durations/) and it's what validation code gives us by default, what do you think?

Reply:
Actually the validation code gives us a `number | Duration | null`, since we deprecated the `sessionTimeout` value in favor of `session.idleTimeout`.
Either way we need a conversion function here, I'm not against using `Duration` instead, but this seemed like the most logical choice to me, since I already had to write the `durationToMs` function to use it elsewhere.

Reply:
> Actually the validation code gives us a number | Duration | null, since we deprecated the sessionTimeout value in favor of session.idleTimeout.

Yeah, but that's because of the way we add `sessionTimeout` into the mix:
```typescript
const sess = config.session;
const session = {
  idleTimeout: (sess && sess.idleTimeout) || config.sessionTimeout || null,
  lifespan: (sess && sess.lifespan) || null,
};

const val = {
  ...config,
  encryptionKey,
  secureCookies,
  session,
};
```
If we did something like this then we'd have `Duration | null` (looking at the schema `session` is always defined):
```typescript
const deprecatedSessionTimeout =
  typeof config.sessionTimeout === 'number' ? duration(config.sessionTimeout) : null;
const val = {
  ...config,
  encryptionKey,
  secureCookies,
  session: {
    ...config.session,
    idleTimeout: config.session.idleTimeout || deprecatedSessionTimeout,
  },
};
```
And with `Duration | null` and optional chaining (:tada:) it seems we don't need a dedicated `durationToMs` utility at all (+ tiny arguable benefit that you immediately see that the value can be `null` while with `durationToMs` it isn't immediately clear):
```typescript
const durationInMsOrNull = config.session.idleTimeout?.asMilliseconds() || null;
```

Having said that I'm not opposed to your solution if you feel strong about it. This code is new to me and I may not see the full picture, just always trying to find a way to avoid having additional utilities :slightly_smiling_face:

Reply:
You've convinced me -- made these changes in 5371f54!